### PR TITLE
Various sanitycheck and testcase fixes

### PR DESCRIPTION
**Pull request by nashif**
- sanitycheck: fail on faults/panics/oopses
- boards: make em_starterkit_em7d default test platform
- sanitycheck: support listing cases not using ztest
- sanitycheck: export list of tests as CSV
- tests: fix test identifiers

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]

File: tests/benchmarks/app_kernel/testcase.yaml [hunk 1]
```diff
@@ -6,7 +8,7 @@ tests:
     tags: benchmark
     slow: true
     timeout: 300
-  test_posix:
+  benchmark.application.posix:
```
Comment (MaureenHelm) [inline tests/benchmarks/app_kernel/testcase.yaml]: These changes don't look related to the script changes. Should they be in the "tests: fix test identifiers" commit instead?

File: tests/benchmarks/object_footprint/testcase.yaml [hunk 2]
```diff
@@ -1,55 +1,55 @@
 tests:
-  test_001:
+  benchmark.footprint.001:
```
Comment (MaureenHelm) [inline tests/benchmarks/object_footprint/testcase.yaml]: These changes don't look related to the script changes. Should they be in the "tests: fix test identifiers" commit instead?

File: scripts/sanity_chk/harness.py [hunk 3]
```diff
@@ -55,6 +59,8 @@ class Test(Harness):
     RUN_PASSED = "PROJECT EXECUTION SUCCESSFUL"
     RUN_FAILED = "PROJECT EXECUTION FAILED"
 
+    faults = ["Unknown Fatal Error", "MPU FAULT", "Kernel Panic", "Kernel OOPS"]
```
Comment (MaureenHelm) [inline scripts/sanity_chk/harness.py]: I think we also need "BUS FAULT" to handle k64 mpu

Comment (nashif) [reply scripts/sanity_chk/harness.py]: done

**Review by MaureenHelm — APPROVED** [review]
